src/commonMain/kotlin/character/classes/rogue/talents/DirtyDeeds.kt

```diff
@@ -0,0 +1,84 @@
+package character.classes.rogue.talents
+
+import character.*
+import sim.SimParticipant
+import data.model.Item
+import sim.Event
+
+// TODO: need to find a better way for this
+
+class DirtyDeeds(currentRank: Int) : Talent(currentRank) {
+    companion object {
+        const val name = "Dirty Deeds"
+    }
+
+    override val name: String = Companion.name
+    override val maxRank: Int = 2
+
+    fun energyReduction(): Double {
+        return currentRank * 10.0
+    }
+
+    fun damageIncreasePercentExecute(): Double {
+        return currentRank * 10.0
+    }
+
+    val executeDmgBuff = object : Buff() {
+        override val name: String = "${Companion.name} (Execute Damage)"
+        override val durationMs: Int = -1
+        override val hidden: Boolean = true
+
+        override fun modifyStats(sp: SimParticipant): Stats {  
+            return Stats(
+                yellowDamageMultiplier = damageIncreasePercentExecute()
+            )
+        }
+    }
+
+    val buff = object : Buff() {
+        override val name: String = "${Companion.name} (Talent)"
+        override val durationMs: Int = -1
+        override val hidden: Boolean = true
+
+        var appliedExecuteBonusBuff = false
```

Comment:
Reworked it with it's own state now. The talent just applies a `Buff`, and this `Buff` applies the actual damage-bonus-buff once the targets health is below 35% (which is checked in the `Proc` that basically procs off everything). This should work now. There might still be issues because `yellowDamageMultiplier` might apply to damage that is not affected by Dirty Deeds.